libraries/rush-lib/src/api/Subspace.ts

```diff
@@ -56,6 +61,23 @@ export class Subspace {
     return this._projects;
   }
 
+  public getPnpmOptions(): PnpmOptionsConfiguration | undefined {
+    if (!this._pnpmOptions) {
+      if (FileSystem.exists(`${this.getSubspaceConfigFolder()}/${RushConstants.pnpmConfigFilename}`)) {
+        try {
+          this._pnpmOptions = PnpmOptionsConfiguration.loadFromJsonFileOrThrow(
+            `${this.getSubspaceConfigFolder()}/${RushConstants.pnpmConfigFilename}`,
+            this.getSubspaceTempFolder()
+          );
+        } catch (e) {
+          throw new Error(`The subspace has an invalid pnpm-config.json file: ${this.subspaceName}`);
+        }
+      }
+    } else {
+      return this._pnpmOptions;
+    }
```

Comment:
```suggestion
    }

    return this._pnpmOptions;
```

Reply:
As written, this function won't return the config file the first time it's called. It'd have to be called 2+ times.

Reply:
changed